src/expt_array_metrics.py

```diff
@@ -345,6 +397,76 @@ def get_expt_record_id(body):
         
     return experiment_id
 
+def get_sat_meta_id_from_metric(metric):
+    sat_meta_id = -1
+    try:    
+        sat_meta_name = metric.sat_meta_name
+        sat_id = metric.sat_id
+        sat_name = metric.sat_name
+        sat_short_name = metric.sat_short_name
+    except Exception as err:
+        print(f'Required sat meta input value not found: {err}')
+        return sat_meta_id
+    
+    if sat_meta_name is None and sat_id is None and sat_name is None and sat_short_name is None:
+        return sat_meta_id
+    
+    sat_meta_request = {
+        'name': 'sat_meta',
+        'method': db_utils.HTTP_GET,
+        'params': {
+            'filters': {
+                'name': {
+                    'exact': sat_meta_name
+                },
+                'sat_name': {
+                    'exact': sat_name
+                },
+                'short_name': {
+                    'exact': sat_short_name
+                },
+                'sat_id': sat_id
+            },
+            'record_limit': 1
+        }
+    }
+
+    print(f'sat_meta_request: {sat_meta_request}')
+
+    smr = SatMetaRequest(sat_meta_request)
+
+    results = smr.submit()
+    print(f'results: {results}')
+
+    record_cnt = 0
+    try:
+        if results.success is True:
+            records = results.details.get('records')
+            if records is None:
+                msg = 'Request for sat meta record did not return a record'
+                raise ExptArrayMetricsError(msg)
+            record_cnt = records.shape[0]
+        else:
+            msg = f'Problems encountered requesting sat meta data.'
+            # create error return db_action_response
+            raise ExptArrayMetricsError(msg)
+        if record_cnt <= 0:
+            msg = 'Request for sat meta record did not return a record'
+            raise ExptArrayMetricsError(msg)
+        
+    except Exception as err:
+        msg = f'Problems encountered requesting sat meta data. err - {err}'
+        raise ExptArrayMetricsError(msg)
+        
+    try:
+        sat_meta_id = records[sm.id.name].iat[0]
+    except Exception as err:
+        error_msg = f'Problem finding sat meta id from record: {records} ' \
+            f'- err: {err}'
+        print(f'error_msg: {error_msg}')
+        raise ExptArrayMetricsError(error_msg) 
```

Comment:
```suggestion
        raise ExptArrayMetricsError(error_msg) from err
```
or replace the above except Exception statement with specific Exception